src/SAML2/XML/saml/Conditions.php

```diff
@@ -0,0 +1,301 @@
+     * @return void
+     */
+    private function setAudienceRestriction(array $audienceRestriction): void
+    {
+        $this->audienceRestriction = $audienceRestriction;
+    }
+
+
+    /**
+     * Collect the value of the oneTimeUse-property
+     *
+     * @return \SAML2\XML\saml\OneTimeUse|null
+     */
+    public function getOneTimeUse(): ?OneTimeUse
+    {
+        return $this->oneTimeUse;
+    }
+
+
+    /**
+     * Set the value of the oneTimeUse-property
+     *
+     * @param \SAML2\XML\saml\OneTimeUse|null $oneTimeUse
+     * @return void
+     */
+    private function setOneTimeUse(?OneTimeUse $oneTimeUse): void
+    {
+        $this->oneTimeUse = $oneTimeUse;
+    }
+
+
+    /**
+     * Collect the value of the proxyRestriction-property
+     *
+     * @return \SAML2\XML\saml\ProxyRestriction|null
+     */
+    public function getProxyRestriction(): ?ProxyRestriction
+    {
+        return $this->proxyRestriction;
+    }
+
+
+    /**
+     * Set the value of the proxyRestriction-property
+     *
+     * @param \SAML2\XML\saml\ProxyRestriction|null $proxyRestriction
+     * @return void
+     */
+    private function setProxyRestriction(?ProxyRestriction $proxyRestriction): void
+    {
+        $this->proxyRestriction = $proxyRestriction;
+    }
+
+
+    /**
+     * Test if an object, at the state it's in, would produce an empty XML-element
+     *
+     * @return bool
+     */
+    public function isEmptyElement(): bool
+    {
+        return (
+            empty($this->notBefore)
+            && empty($this->notOnOrAfter)
+            && empty($this->condition)
+            && empty($this->audienceRestriction)
+            && empty($this->oneTimeUse)
+            && empty($this->proxyRestriction)
+        );
+    }
+
+
+    /**
+     * Convert XML into a Conditions object
+     *
+     * @param \DOMElement $xml The XML element we should load
+     * @return self
+     * @throws \InvalidArgumentException if the qualified name of the supplied element is wrong
+     */
+    public static function fromXML(DOMElement $xml): object
+    {
+        Assert::same($xml->localName, 'Conditions');
+        Assert::same($xml->namespaceURI, Conditions::NS);
+
+        $notBefore = self::getAttribute($xml, 'NotBefore', null);
+        $notOnOrAfter = self::getAttribute($xml, 'NotOnOrAfter', null);
+
+        $condition = Condition::getChildrenOfClass($xml);
+        $audienceRestriction = AudienceRestriction::getChildrenOfClass($xml);
+        $oneTimeUse = OneTimeUse::getChildrenOfClass($xml);
+        $proxyRestriction = ProxyRestriction::getChildrenOfClass($xml);
+
+        return new self(
+            $notBefore !== null ? Utils::xsDateTimeToTimestamp($notBefore) : null,
+            $notOnOrAfter !== null ? Utils::xsDateTimeToTimestamp($notOnOrAfter) : null,
+            $condition,
+            $audienceRestriction,
+            array_pop($oneTimeUse),
+            array_pop($proxyRestriction)
```

Comment:
I'm thinking about this strategy here... The schema allows for multiple elements, but then the standard explicitly forbids it. Now, the question is: should we just ignore multiple elements and pick the last one (as in this case), or should we raise an error?

@thijskh what's your opinion on this? I believe in other places the strategy has been to raise an error, but now I'm in doubt...

Reply:
I think our strategy has been to be less strict on the receiving end and strict on what we generate

Reply:
Yeah, I remember discussing about that, and it definitely makes sense!